Loop_backend/database/idea.go

```diff
@@ -0,0 +1,156 @@
+    "net/http"
+    "your-project/models"
+    "github.com/gin-gonic/gin"
+    "github.com/lib/pq"
+)
+
+type IdeaHandler struct {
+    db *sql.DB
+}
+
+func NewIdeaHandler(db *sql.DB) *IdeaHandler {
+    return &IdeaHandler{db: db}
+}
+
+func (h *IdeaHandler) CreateIdea(c *gin.Context) {
+    var req models.CreateIdeaRequest
+    if err := c.ShouldBindJSON(&req); err != nil {
+        c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request body"})
+        return
+    }
+
+    var idea models.Idea
+    err := h.db.QueryRow(`
+        INSERT INTO ideas (title, description)
+        VALUES ($1, $2)
+        RETURNING id, title, description, created_at
+    `, req.Title, req.Description).Scan(
+        &idea.ID,
+        &idea.Title,
+        &idea.Description,
+        &idea.CreatedAt,
+    )
+
+    if err != nil {
+        if pqErr, ok := err.(*pq.Error); ok {
+            // Handle specific PostgreSQL errors
+            switch pqErr.Code {
+            case "23505": // unique_violation
+                c.JSON(http.StatusConflict, gin.H{"error": "Idea already exists"})
+                return
+            default:
+                c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to create idea"})
+                return
+            }
+        }
+        c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to create idea"})
+        return
+    }
+
+    c.JSON(http.StatusCreated, idea)
+}
+
+func (h *IdeaHandler) GetIdeas(c *gin.Context) {
+    rows, err := h.db.Query(`
+        SELECT id, title, description, created_at
+        FROM ideas
+        ORDER BY created_at DESC
+    `)
+    if err != nil {
+        c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to fetch ideas"})
+        return
+    }
+    defer rows.Close()
+
+    var ideas []models.Idea
+    for rows.Next() {
+        var idea models.Idea
+        if err := rows.Scan(
+            &idea.ID,
+            &idea.Title,
+            &idea.Description,
+            &idea.CreatedAt,
+        ); err != nil {
+            c.JSON(http.StatusInternalServerError, gin.H{"error": "Failed to process ideas"})
+            return
+        }
+        ideas = append(ideas, idea)
+    }
+
+    if err = rows.Err(); err != nil {
+        c.JSON(http.StatusInternalServerError, gin.H{"error": "Error while iterating ideas"})
+        return
+    }
+
+    c.JSON(http.StatusOK, ideas)
+}
+
+// main.go
+package main
+
+import (
+    "database/sql"
+    "log"
+    "your-project/handlers"
+    "github.com/gin-gonic/gin"
+    "github.com/lib/pq"
+)
+
+func main() {
+    // Database connection
+    db, err := sql.Open("postgres", "postgres://username:password@localhost:5432/dbname?sslmode=disable")
+    if err != nil {
+        log.Fatal(err)
+    }
+    defer db.Close()
+
+    if err = db.Ping(); err != nil {
+        log.Fatal(err)
+    }
+
+    r := gin.Default()
+
+    ideaHandler := handlers.NewIdeaHandler(db)
+
+    r.POST("/api/ideas", ideaHandler.CreateIdea)
```

Comment:
Do not create a new file for ideaBoard. Use the existing backend code and and API routes. Check the working of Projects page.